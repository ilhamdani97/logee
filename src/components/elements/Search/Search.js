import React from 'react';
import styles from './styles.scoped.css';
import PropTypes from 'prop-types';
import Button from '../Button';

export default function Search(props) {
  const { handleSubmit,changeKeyword,keyword,disabled,inputProps, className } = props;
  const classes = [styles.root, className].filter(Boolean).join(' ');

  const buttonHandleSubmit = (e) => {
    e.preventDefault();
    handleSubmit(keyword);
  };

  const getPlaceholder = () => {
    if(inputProps.placeholder && inputProps.placeholder.length > 21){
      return `${inputProps.placeholder.substring(0, 22)}...`;
    } else {
      return inputProps.placeholder;
    }
  };

  return(
    <div className={classes}>
      <form onSubmit={buttonHandleSubmit}>
        <div>
          <input disabled={disabled} onChange={(e)=> changeKeyword(e.target.value)}
            {...inputProps} placeholder={getPlaceholder()} value = {keyword}/>
          <div>
            <Button buttonProps={{ type: 'submit' }}>
              <img src="/assets/search.svg"/>
            </Button>
          </div>
        </div>
      </form>
    </div>
  );
}

Search.defaultProps = {
  changeKeyword : () => {},
  className: '',
  disabled:false,
  handleSubmit  : () => {},
  inputProps: {
    placeholder: 'Cari'
  },
  keyword : '',
};

Search.propTypes = {
  changeKeyword: PropTypes.func,
  className: PropTypes.string,
  disabled:PropTypes.bool,
  handleSubmit: PropTypes.func,
  inputProps: PropTypes.object,
  keyword: PropTypes.string
};
