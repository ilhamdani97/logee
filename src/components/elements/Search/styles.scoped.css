.root > form > div {
  background-color: var(--background-color-2);
  border-radius: 0.1875rem;
  width: 15rem;
  border: 0.125rem;
  border-color: var(--grey-color-2);
  box-sizing: border-box;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 2rem;
  width: 15rem;
}

.root > form > div > input {
  border: 0.0625rem solid var(--grey-color-3);
  border-radius:  0.25rem;
  box-shadow: none;
  color: var(--text-color);
  display: block;
  font-size: 0.875rem;
  font-weight: bold;
  padding: 0.75rem 0.5rem;
  width: 13.5rem;
  height: 2.125rem;
  background-color: var(--background-color-2);
  transition: border 0.25s;
  overflow: auto;
  padding-right: 2.8rem;
}

.root > form > div > input:disabled {
  background-color: var(--background-color);
}

.root > form > div > input:focus {
  border-color: var(--primary-color);
}

.root > form > div > input::placeholder {
  color: var(--light-text-color);
}

.root > form > div > div {
  height: auto;
  margin-left: -2.1rem;
}

.root > form > div > div > button {
  height: 2rem;
  width: 2.05rem;
  background-color: var(--light-text-color);
  border-radius: 0 0.23rem 0.23rem 0;
  display: flex;
  justify-content: center;
  align-items: center;
  border-left: 1px solid var(--grey-color-3);
}

.root > form > div > div > button > img {
  height: 0.910626rem;
  width: 0.910625rem;
}


