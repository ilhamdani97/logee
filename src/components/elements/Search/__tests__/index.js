import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Search from '../Search';

describe('src/components/fields/Search', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Search
      changeKeyword={jest.fn}
      handleSubmit={jest.fn}
      keyword="keyword"
    />);
    expect(tree).toMatchSnapshot();
    const inputField = tree.props.children.props.children.props.children[0];
    tree.props.children.props.onSubmit({
      preventDefault: jest.fn()
    });
    inputField.props.onChange({
      target: {
        value: ''
      }
    });
    Search.defaultProps.handleSubmit();
    Search.defaultProps.changeKeyword();
  });
});
