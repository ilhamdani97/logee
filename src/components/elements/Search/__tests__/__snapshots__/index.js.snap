// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/fields/Search render 1`] = `
<div
  className=""
>
  <form
    onSubmit={[Function]}
  >
    <div>
      <input
        disabled={false}
        onChange={[Function]}
        placeholder="Cari"
        value="keyword"
      />
      <div>
        <Button
          buttonProps={
            Object {
              "type": "submit",
            }
          }
          className=""
          disabled={false}
          isLoading={false}
          onClick={[Function]}
        >
          <img
            src="/assets/search.svg"
          />
        </Button>
      </div>
    </div>
  </form>
</div>
`;
